beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs

```diff
@@ -0,0 +1,472 @@
+            InboundEvent::Msg(EarlyBlock(early_block)) => {
+                let block_slot = early_block.block.block.slot();
+                let block_root = early_block.block.block_root;
+
+                // Don't add the same block to the queue twice. This prevents DoS attacks.
+                if self.queued_block_roots.contains(&block_root) {
+                    return;
+                }
+
+                if let Some(duration_till_slot) = slot_clock.duration_to_slot(block_slot) {
+                    // Check to ensure this won't over-fill the queue.
+                    if self.queued_block_roots.len() >= MAXIMUM_QUEUED_BLOCKS {
+                        error!(
+                            log,
+                            "Early blocks queue is full";
+                            "queue_size" => MAXIMUM_QUEUED_BLOCKS,
+                            "msg" => "check system clock"
+                        );
+                        // Drop the block.
+                        return;
+                    }
+
+                    self.queued_block_roots.insert(block_root);
+                    // Queue the block until the start of the appropriate slot, plus
+                    // `ADDITIONAL_QUEUED_BLOCK_DELAY`.
+                    self.block_delay_queue.insert(
+                        early_block,
+                        duration_till_slot + ADDITIONAL_QUEUED_BLOCK_DELAY,
+                    );
+                } else {
+                    // If there is no duration till the next slot, check to see if the slot
+                    // has already arrived. If it has already arrived, send it out for
+                    // immediate processing.
+                    //
+                    // If we can't read the slot or the slot hasn't arrived, simply drop the
+                    // block.
+                    //
+                    // This logic is slightly awkward since `SlotClock::duration_to_slot`
+                    // doesn't distinguish between a slot that has already arrived and an
+                    // error reading the slot clock.
+                    if let Some(now) = slot_clock.now() {
+                        if block_slot <= now
+                            && self
+                                .ready_work_tx
+                                .try_send(ReadyWork::Block(early_block))
+                                .is_err()
+                        {
+                            error!(
+                                log,
+                                "Failed to send block";
+                            );
+                        }
+                    }
+                }
+            }
+            InboundEvent::Msg(UnknownBlockAggregate(queued_aggregate)) => {
+                if self.attestations_delay_queue.len() >= MAXIMUM_QUEUED_ATTESTATIONS {
+                    error!(
+                        log,
+                        "Attestation delay queue is full";
```

Comment:
```suggestion
                        "Aggregate attestation delay queue is full";
```

Just to help us disambiguate between this and the unaggregate errors.